includes/class-wc-amazon-payments-advanced-ipn-handler.php

```diff
@@ -408,10 +408,31 @@ public function handle_notification_ipn_v2( $message ) {
 		$notification_version = isset( $message['Message']['NotificationVersion'] ) ? strtolower( $message['Message']['NotificationVersion'] ) : 'v1';
 
 		if ( 'v2' !== $notification_version ) {
-			return;
-		}
+			$notification_data = self::safe_load_xml( $message['Message']['NotificationData'], LIBXML_NOCDATA );
+
+			switch ( $message['Message']['NotificationType'] ) {
+				case 'PaymentCapture':
+					$type = 'CHARGE';
+					$charge_id = (string) $notification_data->CaptureDetails->AmazonCaptureId;
+					break;
+				case 'PaymentRefund':
+					$type = 'REFUND';
+					$charge_id     = (string) $notification_data->RefundDetails->AmazonRefundId;
+					break;
+				default:
+					wc_apa()->log( 'No handler for notification with type ' . $message['Message']['NotificationType'] );
+					return;
+			}
 
-		$notification = $message['Message'];
+			$notification  = array(
+				'NotificationVersion' => 'V1',
+				'NotificationType'    => 'STATE_CHANGE',
+				'ObjectType'          => $type,
+				'ObjectId'            => $charge_id,
+			);
```

Comment:
Should we log that we're transforming a v1 received IPN to be processed as v2? Just for reference.